mobile/src/androidTest/java/com/novoda/canvas/BounceTest.java

```diff
@@ -0,0 +1,152 @@
+                                parent.addView(ball);
+                                throwBallForward(starIndex % 2 == 0, ball, parent);
+                                tossBallUp(ball, parent);
+                            }
+                        }
+                    }
+                    , batchIndex * 500
+            );
+        }
+
+    }
+
+    private List<ImageView> createObjects(int count, Activity activity, ViewGroup parent) {
+        ArrayList<ImageView> objects = new ArrayList<>(count);
+
+        for (int i = 0; i < count; i++) {
+            ImageView object = new ImageView(activity);
+            int size = getBallSize();
+            Bitmap circle = createCircle(activity.getResources(), size, size);
+            object.setImageBitmap(circle);
+            object.setLayoutParams(new ActionBar.LayoutParams(size, size));
+            object.setX(RANDOM.nextInt(parent.getWidth()));
+            object.setY(parent.getHeight());
+            objects.add(object);
+        }
+        return objects;
+    }
+
+    private Bitmap createCircle(Resources resources, int width, int height) {
+        final Bitmap output = Bitmap.createBitmap(
+                width, height, Bitmap.Config.ARGB_8888
+        );
+        final Canvas canvas = new Canvas(output);
+
+        final int color = getBallColor();
+        final Paint paint = new Paint();
+        final Rect rect = new Rect(0, 0, width, height);
+        final RectF rectF = new RectF(rect);
+
+        paint.setAntiAlias(true);
+        canvas.drawARGB(0, 0, 0, 0);
+        paint.setColor(resources.getColor(COLOR_VARIANTS[color]));
+        canvas.drawOval(rectF, paint);
+
+        paint.setXfermode(new PorterDuffXfermode(PorterDuff.Mode.SRC_IN));
+
+        return output;
+    }
+
+    private void throwBallForward(boolean leftToRight, ImageView ball, ViewGroup parent) {
+        float start = getDropPoint();
+        float end = parent.getWidth() + ball.getWidth();
+        if (!leftToRight) {
+            float temp = start;
+            start = end;
+            end = temp;
+        }
+
+        ObjectAnimator tossForward = ObjectAnimator.ofFloat(ball, "x", start, end);
+        tossForward.setDuration(forwardTossDuration());
+        tossForward.setInterpolator(new AccelerateDecelerateInterpolator());
+        tossForward.start();
+
+    }
+
+    private void tossBallUp(ImageView ball, ViewGroup parent) {
+        ObjectAnimator tossUp = ObjectAnimator.ofFloat(ball, "y", getDropPoint(), parent.getHeight() - MAX_BALL_SIZE);
+        tossUp.setDuration(tossDuration());
+        tossUp.setInterpolator(new BounceInterpolator());
+        tossUp.start();
+    }
+
+    private int getDropPoint() {
+        return -(RANDOM.nextInt(500) + MAX_BALL_SIZE);
+    }
+
+    private int getBallSize() {
+        return RANDOM.nextInt(MAX_BALL_SIZE - 10) + 10;
+    }
+
+    private int getBallColor() {
+        return RANDOM.nextInt(COLOR_VARIANTS.length);
```

Comment:
The colors are not that random really. They are the base colors from https://www.google.com/design/spec/style/color.html which all work on a dark background.

Then again I didn't really know about that class :P